DEVELOPING.md

```diff
@@ -0,0 +1,62 @@
+# Requirements
+* Python 3.8+
+* Poetry (`curl -sSL https://install.python-poetry.org | python - --version=1.2.2`)
+
+For the other requirements, inspect the ``pyproject.toml`` file. If you are updated the dependencies, please run `poetry update` to update the
+
+# Development Tasks
+There are a series of top-level tasks available through Poetry. These can each be run via
+
+ `poetry run poe <taskname>`
+
+### Basic Verification
+* **apply_format** - runs the suite of formatting tools applying tools to make code compliant
+* **check_format** - runs the suite of formatting tools checking for compliance
+* **lint** - runs the suite of linting tools
+* **typecheck** - performs static typechecking of the codebase using mypy
+* **unit_test** - executes fast unit tests
+* **verify** - executes the basic PR verification suite, which includes all the tasks listed above
```

Comment:
I am missing "format" here, i.e., `poetry run poe format`. This is actually all I do normally.

Strictly speaking, this is the only thing that is necessary (assuming that the tests are working).

Reply:
`poetry run poe format` is the same as `poetry run poe apply_format`. I suppose we should actually consolidate this across all pywhy packages, so the dev exp is as similar as possible

Reply:
Okay changed.